docs/datastore/neo4j_graph.md

```diff
@@ -0,0 +1,94 @@
+# Setup and configure Neo4j
+
+## Before you begin
+
+1. Make sure you have a GoogleCloud project and billing is enabled.
+
+1. Set your `PROJECT_ID` environment variable:
+
+    ```bash
+    export PROJECT_ID=<YOUR_PROJECT_ID>
+    ```
+
+1. [Install](https://cloud.google.com/sdk/docs/install) the gcloud CLI.
+
+1. Set gcloud project:
+
+    ```bash
+    gcloud config set project $PROJECT_ID
+    ```
+
+1. Make sure you have python version 3.11+ installed.
+
+    ```bash
+    python -V
+    ```
+
+1. [Install python][install-python] and set up a python [virtual environment][venv].
+
+1. Install required dependencies:
+    ```bash
+    cd retrieval_service
+    pip install -r requirements.txt
+    ```
+
+## Create a Neo4j database
+
+1. Important: Enabling Neo4j Aura in Google Cloud Marketplace does not permit the use of `Free-tier` instances. It will provision a `Professional-tier` instance, which incurs additional monthly costs.
```

Comment:
Let's create two sub-sections and have instructions under the corresponding section. Also for any attached link 

### Neo4j AuraDB on GCP
you can choose not to include link for this option as it is auto-directed and has a lot of extra parameters. The link that you currently attached has a bunch of session data that shouldn't be included.

### Neo4j Aura (free)
show steps and link https://console.neo4j.io/ to this option